### ISSUE TITLE
[Technical Research] Net new environment for EEC

### ISSUE THREAD

**Issue by eeeady** [OHS-Hosting-Infrastructure/infrastructure#30]
Get an understanding of the new EEC environment for the ICF team.

My understanding of what we'll need is:
RDS instance with MySql (?), ASG (with a single instance to start), ALB + non production DNS/Certs. Please validate this and spin out new tickets.

What we need to know from them is:
- Do they already have an AMI?
- Do they have CI?
- Do they have a repo?

AC:
- [ ] Design of CI flow
- [ ] Architecture diagram for new env
- [ ] New repos for collaborating on AMI build and Terraform of environments

**Comment by robwilkerson** [comment]
@eeeady - in no particular order...

- I'm hoping to move forward with a containerized solution deployed to Fargate (or EKS, but Fargate just feels simpler and more applicable for this)
- RDS, definitely, but might use MariaDB rather than MySQL
- Would def like certs + dns (dev., stage., etc.) for lower envs
- We're standardizing on Travis for CI, but have no template in place just yet
- We'd love to CD to dev, then push-button deploy to stage + prod
- We will have a repo, but don't just yet; I think we're tasked with hosting this, so it would exist in the [@ICF-Next-Government](https://github.com/ICF-Next-Government) organization
- I suspect that ops code would make the most sense in its own `eec-operations` repo containing any Terraform/Ansible/operations scripts

I have some generic path-to-production diagrams in place that i can drop here as concepts, but the actual implementation may vary based on project needs.

TL;DR: if you're asking for technical stuff we already have specific to this project, it doesn't exist - we're not really any further ahead than you are on that since we're still very much in discovery mode.

Tag me if you have specific questions and we'll keep the conversations rolling here rather than emails...ideally. :-)

**Comment by robwilkerson** [comment]
Also, this will be a Drupal project so it will require mounted asset storage. I'm not entirely sure how it can/would work with a containerized solution, but it feels handy to engage cloudfront/cdn capabilities rather than an EFS mount if we can sort out the implementation.

**Comment by eeeady** [comment]
Thanks for the additional info @robwilkerson !

I forgot to ask, do you have a timeline for completion? I assume ASAP but we need a realistic sense of when we should target at least having a dev CD flow working since I'm under the impression your engagement is focused on the upgrade to Drupal 9 by the end of the year.

**Comment by robwilkerson** [comment]
I'm honestly not sure about the timeline, @eeeady . Let me see if I can nail that down a bit.

**Comment by robwilkerson** [comment]
do we have param store access so we can start putting some config values in place, by any chance?

**Comment by eeeady** [comment]
Hmm... We haven't started using parameter store but are 💯 on board with using it. We'd need to get your access set up to do so.

**Comment by dan-gorman-ICF** [comment]
@eeeady - hope you had a good weekend. wanted to make sure you saw my email from Friday - would it be possible to get on a call early this week to 100% align on the infrastructure work allocation for Truss and for ICF? Thanks!

**Comment by rpdelaney** [comment]
Since we have moved work tracking to JIRA, I am closing open issues in this repo via a script.
